arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,241 @@
+                    case .polyline:
+                        let polyline = selectedFeature.geometry as! AGSPolyline
+                        // If the selected feature is a polyline with any part containing more than one segment (i.e. a curve), present an alert.
+                        if polyline.parts.array().contains(where: { $0.pointCount > 2 }) {
+                            self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                            return
+                        }
+                    default:
+                        return
+                    }
+                    // Select the feature.
+                    featureLayer.select(selectedFeature)
+                    // Set the feature globally.
+                    self.selectedFeature = selectedFeature
+                    return
+                }
+            }
+        }
+    }
+    
+    // Create an alert dialog with edit texts to allow editing of the given feature's 'AD_ADDRESS' and 'ST_STR_NAM' attributes.
+    func showEditableAttributes(selectedFeature: AGSFeature) {
+        // Create an object to observe changes from the text fields.
+        var textFieldObserver: NSObjectProtocol!
+        // Create an alert controller and customize the title and message.
+        let alert = UIAlertController(title: "Edit Feature Attributes", message: "Edit the 'AD_ADDRESS' and 'ST_STR_NAM' attributes.", preferredStyle: .alert)
+        // Add a "Done" option to complete the editing process and close the alert.
+        let doneAction = UIAlertAction(title: "Done", style: .default) { _ in
+            let addressTextField = alert.textFields?[0]
+            let streetTextField = alert.textFields?[1]
+            // Set the new values for the attributes and update the selected feature.
+            selectedFeature.attributes["AD_ADDRESS"] = Int((addressTextField?.text)!)
+            selectedFeature.attributes["ST_STR_NAM"] = streetTextField?.text
+            selectedFeature.featureTable?.update(selectedFeature)
+            // Remove the observer after editing is complete.
+            if let textFieldObserver = textFieldObserver {
+                NotificationCenter.default.removeObserver(textFieldObserver)
+            }
+        }
+        alert.addAction(doneAction)
+        // Add a text field to prompt the user to input the street address.
+        alert.addTextField { (textField) in
+            // Prompt a number pad for the address.
+            textField.keyboardType = .numberPad
+            // Populate the text fields with the current value.
+            textField.text = (selectedFeature.attributes["AD_ADDRESS"] as! NSNumber).stringValue
+            // Add an observer to ensure the user does not input an empty string.
+            textFieldObserver = NotificationCenter.default.addObserver(forName: UITextField.textDidChangeNotification, object: nil, queue: .main, using: {_ in
```

Comment:
Since the observer is no longer tied to a specific text field, it no longer needs to be in the configuration handler for this text field. I would move it out of the closure (into the enclosing method, where `textFieldObserver` is declared).